platforms/android/library-compose/src/androidTest/java/io/element/android/wysiwyg/compose/RichTextEditorStateTest.kt

```diff
@@ -54,6 +57,37 @@ class RichTextEditorStateTest {
         onView(withText("Hello, world")).check(matches(isDisplayed()))
     }
 
+    @Test
+    fun testStateRestoration() = runTest {
+        val state = RichTextEditorState()
+        val hideEditor = MutableStateFlow(false)
+        composeTestRule.setContent {
+            MaterialTheme {
+                val hide by hideEditor.collectAsState()
+                Column() {
+                    if (!hide) {
+                        Text("Editor")
+                        RichTextEditor(modifier = Modifier.fillMaxWidth(), state = state)
+                    }
+                }
+            }
+        }
+
+        state.setHtml("Hello, world")
+        // Ensure line count is set
+        Assert.assertEquals(1, state.lineCount)
+
+        // Hide and show the editor to simulate a configuration change
+        hideEditor.emit(true)
+        hideEditor.emit(false)
+        composeTestRule.awaitIdle()
+
+        // If the text is found, the state was restored
+        onView(withText("Hello, world")).check(matches(isDisplayed()))
+        // Line count is kept
+        Assert.assertEquals(1, state.lineCount)
```

Comment:
There is something a bit odd going on with the line count now; adding this assertion to the test above (`testSharingState`) causes it to fail with `state.lineCount == 12`. Shall we add add the line count check to that test and if possible get it returning the correct result?

Reply:
This is because the `EditText` view is way too small and it ends up wrapping the text, I added a `Modifier.fillMaxWidth()` to the other test to fix this same issue, I can add it there too.

Reply:
Oh of course, that makes sense. Thank you!